packages/react-components/react-components/src/AccessibilityScenarios/Input.stories.tsx

```diff
@@ -305,24 +318,106 @@ const RegistrationFormInputsAccessibility = () => {
             </ValidationMessage>
           )}
 
+          <Label htmlFor="birthDate">Birth date:</Label>
+          <Controller
+            name="birthDate"
+            control={control}
+            as={
+              <Input
+                type="text"
+                id="birthDate"
+                placeholder="E.g. 3/21/1995"
+                aria-required="true"
+                aria-invalid={!!errors.birthDate}
+                aria-describedby="birthDateErrors"
+              />
+            }
+            rules={{
+              required: true,
+              validate: {
+                validDate: value => regexes.validDate.test(value),
+                always: () => {
+                  if (!formState.isSubmitting) {
+                    formValidation.onFieldValidated('birthDate');
+                  }
+                  return true;
+                },
+              },
+            }}
+          />
+          {errors.birthDate?.types && (
+            <ValidationMessage id="birthDate" formValidation={formValidation}>
+              {'required' in errors.birthDate.types ? (
+                <p>Birth date is required.</p>
+              ) : (
+                <>
+                  <p>Birth date is invalid. It must:</p>
+                  <ul>{'validDate' in errors.birthDate.types && <li>Be in the MM/DD/YYYY format.</li>}</ul>
+                </>
+              )}
+            </ValidationMessage>
+          )}
+
+          <Checkbox label="Send me newsletter" onChange={onSendNewsletterChange} />
+
+          <Label htmlFor="email">E-mail:</Label>
+          <Controller
+            name="email"
+            control={control}
+            as={
+              <Input
+                type="text"
+                id="email"
+                disabled={!isSendNewsletter}
```

Comment:
Similar comment as before, this only applies if we're exposing these scenarios on our docs site for authors:

as a general usability best practice, I think it'd be better to show/hide this field based on `!isSendNewsletter` rather than toggling disabled state.

Reply:
I agree in the principle. Though as the purpose of these scenarios is at the first place to present to the testers all the different variants and types of the components in question, I have been a bit struggling to find a meaningful example of all these variants and types. Please, do you have a better idea for a disabled input to demonstrate in this scenario?

Reply:
I think the current use case is good for testing, and it's probably better to leave them as-is for now and run tests to get feedback sooner than later. Then maybe we can revisit this if we want to repurpose the examples for showing authors.

Reply:
Okay, anyway, do you please have a good idea for a disabled input example? It does not necessary hav to be part of this scenario. I could create a new scenario for that.